### ISSUE TITLE
where to set "Cross attention optimization" Does  comfyui have "Negative Guidance minimum sigma"

### ISSUE THREAD

**Issue by xueqing0622** [comfyanonymous/ComfyUI#1891]
where to set "Cross attention optimization" Does  comfyui have "Negative Guidance minimum sigma"

https://stable-diffusion-art.com/speed-up-stable-diffusion/
![Image text](https://stable-diffusion-art.com/wp-content/uploads/2023/07/image-1-1024x549.png)

![Image text](https://stable-diffusion-art.com/wp-content/uploads/2023/07/image-3-1024x432.png)

**Comment by NeedsMoar** [comment]
You set cross attention on the command line.  xformers is the default if installed and on nvidia, if you want different you can specify the other options (it'll disable xformers) or pass in --disable-xformers and let comfy decide (it should go to pytorch, at least on nvidia.  Directml defaulted to quad last I used it):

--use-split-cross-attention
--use-quad-cross-attention
--use-pytorch-cross-attention

[smZNodes](https://github.com/shiimizu/ComfyUI_smZNodes) has a node called "Settings (smZ)" which has the negative minimum sigma option and the pad prompt / negative prompt options from that menu as well as many others.

**Comment by NeedsMoar** [comment]
I'd also suggest trying out the "HyperTiling" node under _nodes_for_testing.  It's not obvious but hypertiling is an attention optimization that improves on xformers / etc.   HyperTiling increases speed as the image size increases.   It's maybe 1.2x faster at 1024x1024 for me and about 2.1x faster at 2048x2048.  According to the original repo it can be up to 4x faster at higher resolutions than that.  There's also TomeSD which is a different animal that removes effectively duplicate keywords from the model and their repo showed a drop from 3.4GB to 0.6GB of vram use and 5.4x speed increase at a strength of 60% and 512x512.  Finally the FreeU nodes from that menu improve the quality of the generated image without affecting generation speed so they're worth trying too. You'll probably get far more performance out of either of the first two there than from the negative minimum sigma.

**Comment by xueqing0622** [comment]
Yes, thanks for the reminder, according to what you said about optimization, there is a good balance between the quality of the images and the speed.
